test/solvers/unconstrained-nls.jl

```diff
@@ -0,0 +1,119 @@
+function test_unconstrained_nls_solver(solver)
+  @testset "Simple tests" begin
+    @testset "f(x) = x₁² + x₂²" begin
+      nls = ADNLSModel(x->x, ones(2), 2)
+
+      stats = with_logger(NullLogger()) do
+        solver(nls)
+      end
+      @test isapprox(stats.solution, zeros(2), atol=1e-6)
+      @test isapprox(stats.objective, 0.0, atol=1e-16)
+      @test stats.dual_feas < 1e-6
+      @test stats.status == :first_order
+    end
+
+    @testset "Diagonal quadratic" begin
+      n = 100
+      D = [0.1 + 0.9 * (i - 1) / (n - 1) for i = 1:n]
+      nls = ADNLSModel(x->sqrt.(D) .* x, ones(n), n)
+
+      stats = with_logger(NullLogger()) do
+        solver(nls)
+      end
+      @test isapprox(stats.solution, zeros(n), atol=1e-6)
+      @test isapprox(stats.objective, 0.0, atol=1e-6)
+      @test stats.dual_feas < 1e-6
+      @test stats.status == :first_order
+    end
+
+    @testset "Rosenbrock" begin
+      nls = ADNLSModel(x -> [x[1] - 1; 10 * (x[2] - x[1]^2)], [-1.2; 1.0], 2)
+
+      stats = with_logger(NullLogger()) do
+        solver(nls, max_eval=-1)
+      end
+      @test isapprox(stats.solution, ones(2), atol=1e-6)
+      @test isapprox(stats.objective, 0.0, atol=1e-6)
+      @test stats.dual_feas < 1e-6
+      @test stats.status == :first_order
+    end
+
+    @testset "Underdeterminded" begin
```

Comment:
Determinded.

Reply:
In this case, `m < n` and there are infinite solutions. Shouldn't be `underdetermined`?

Reply:
I mean that you wrote "determinded" instead of "determined".

Reply:
(And so did I, I now realize... sorry)

Reply:
Ah, sorry. Will fix